skrub/_joiner.py

```diff
@@ -45,66 +130,70 @@ class Joiner(TransformerMixin, BaseEstimator):
+        ``main_table`` and their nearest neighbor in ``aux_table`` will be
+        rescaled by this reference distance.
+    string_encoder : scikit-learn transformer used to vectorize text columns
+        By default a ``HashingVectorizer`` combined with a ``TfidfTransformer``
+        is used.
+    insert_match_info : bool, default=True
+        Insert some columns whose names start with `skrub.Joiner` containing
+        the distance, rescaled distance and whether the rescaled distance is
+        above the threshold.
 
     See Also
     --------
     AggJoiner :
         Aggregate auxiliary dataframes before joining them on a base dataframe.
 
     fuzzy_join :
-        Join two tables (dataframes) based on approximate column matching.
-
-    get_ken_embeddings :
-        Download vector embeddings for many common entities (cities,
-        places, people...).
+        Join two tables (dataframes) based on approximate column matching. This
+        is the same functionality as provided by the ``Joiner`` but exposed as
+        a function rather than a transformer.
 
     Examples
     --------
-    >>> X = pd.DataFrame(['France', 'Germany', 'Italy'], columns=['Country'])
-    >>> X
-       Country
-    0   France
-    1  Germany
-    2    Italy
-
-    >>> aux_table = pd.DataFrame([['germany', 84_000_000],
-    ...                         ['france', 68_000_000],
-    ...                         ['italy', 59_000_000]],
-    ...                         columns=['Country', 'Population'])
+    >>> import pandas as pd
+    >>> main_table = pd.DataFrame({"Country": ["France", "Italia", "Spain"]})
+    >>> aux_table = pd.DataFrame( {"Country": ["Germany", "France", "Italy"],
+    ...                            "Capital": ["Berlin", "Paris", "Rome"]} )
+    >>> main_table
+      Country
+    0  France
+    1  Italia
+    2   Spain
     >>> aux_table
-       Country  Population
-    0  germany    84000000
-    1   france    68000000
-    2    italy    59000000
-
-    >>> joiner = Joiner(aux_table, key='Country', suffix='_aux')
-
-    >>> augmented_table = joiner.fit_transform(X)
-    >>> augmented_table
-       Country Country_aux  Population
-    0   France      france    68000000
-    1  Germany     germany    84000000
-    2    Italy       italy    59000000
+       Country Capital
+    0  Germany  Berlin
+    1   France   Paris
+    2    Italy    Rome
+    >>> joiner = Joiner(
+    ...     aux_table,
+    ...     key="Country",
+    ...     suffix="_capitals",
+    ...     max_dist=0.9,
+    ...     insert_match_info=False,
+    ... )
+    >>> joiner.fit_transform(main_table)
+      Country Country_capitals Capital_capitals
+    0  France           France            Paris
+    1  Italia            Italy             Rome
+    2   Spain              NaN              NaN
     """
 
+    _match_info_keys = ["distance", "rescaled_distance", "match_accepted"]
+    _match_info_key_renaming = {k: f"skrub.Joiner.{k}" for k in _match_info_keys}
```

Comment:
I usually avoid accessing columns as attributes but I agree some users might; in any case "_" instead of "." doesn't have any drawbacks.

In general I would like to have something a bit more expressive like a format string or at least a prefix rather than a suffix to indicate which table a column came from, because usually (except domain names) we start with more general and end with more specific so `"World_Bank.Table_1.{}"` seems better than `"{}_World_Bank_Table_1"` for formatting column names IMO

Reply:
100% agree with you